src/it/resources/com/google/checkstyle/test/chapter5naming/rule523methodnames/InputMethodName.java

```diff
@@ -42,6 +42,17 @@ void testingFoo_foo() {}
   @Test
   void testing_0123() {}
 
+  @Test
+  void Testing_Foo() {} // violation 'Method name 'Testing_Foo' must match pattern'
+
+  @Test
+  void t_esting() {} // violation 'Method name 't_esting' must match pattern'
+
+  @Test
+  void _testing() {} // violation 'Method name '_testing' must match pattern'
```

Comment:
Now we're able to detect `Testing_Foo`. We're giving violation for `t_esting` and `_testing`, there's no explicit rule specified for placement of `_` in test method name, the initial character of test method is also lowercase character so ideally we should not give violation for these methods but no one writes names like this and no one should actually do, I think we should keep this violation but we should add blue tick for MethodName module on coverage page

Reply:
yes, such crazy names are ok to violate.